### PR TITLE
update amhswagger and add folder structure

### PR DESCRIPTION
**Pull request by athulvijayanmsft**
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by athulvijayanmsft** [comment]
> @athulvijayanmsft please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

File: specification/migrate/resource-manager/readme.md
Comment (tarsharm) [inline specification/migrate/resource-manager/readme.md:169]: >MigrateProjects [](http://example.com/codeflow?start=20&length=15)

@athulvijayanmsft the folder name is decided as MIgrateProjects?

**Comment by rkmanda** [comment]
> Please address or respond to feedback from the ARM API reviewer. When you are ready to continue the ARM API review, please remove the `ARMChangesRequested` label. This will notify the reviewer to have another look. If the feedback provided needs further discussion, please use this Teams channel to post your questions - https://aka.ms/azsdk/support/specreview-channel. Please indicate that this is an ARM-related question in the title of your post by including "[ARM Query]" in the title.

Please address the breaking changes and once you have either fixed the PR or gotten a breaking change approval, you can remove the ARMCHangesRequested label from the PR to continue the review

**Comment by ms-henglu** [comment]
Please check and fix errors: https://github.com/Azure/azure-rest-api-specs/pull/24162/checks?check_run_id=16066130798

**Comment by athulvijayanmsft** [comment]
> Please check and fix errors: https://github.com/Azure/azure-rest-api-specs/pull/24162/checks?check_run_id=16066130798

Can you continue review the rest while I fix the prettier checks? as we are reaching deadline for completion.

**Comment by athulvijayanmsft** [comment]
> Please address or respond to feedback from the ARM API reviewer. When you are ready to continue the ARM API review, please remove the `ARMChangesRequested` label. This will notify the reviewer to have another look. If the feedback provided needs further discussion, please use this Teams channel to post your questions - [aka.ms/azsdk/support/specreview-channel](https://aka.ms/azsdk/support/specreview-channel). Please include `[ARM Query]` in the title of your question to indicate that it is ARM-related.

@rkmanda 
fixed prettier errors and prettier check is passing locally for all the files; but the PR check still fails. Please unblock here.
Please queue this for further review

**Comment by athulvijayanmsft** [comment]
> Please check and fix errors: https://github.com/Azure/azure-rest-api-specs/pull/24162/checks?check_run_id=16066130798

Resolved and checks are passing now.

**Comment by sjanamma** [comment]
>           "nextLinkName": null

Pagination needs to be supported for the list calls. nextLinkName cannot be null

---
Refers to: specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json:224 in 20fefa5. [](commit_id = 20fefa52c90f4a6e9791b2d5934dc071122e46ce, deletion_comment = False)

**Comment by mentat9** [comment]
@athulvijayanmsft - Your PR needs to pass checks and address the SDK breaking change labels before merge. Add the MergeRequested label back after addressing those items. If you get stuck, you can ask for help in the API Spec Review Teams channel: https://teams.microsoft.com/l/channel/19%3a0351f5f9404446e4b4fd4eaf2c27448d%40thread.skype/API%2520Spec%2520Review?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47.

**Comment by kazrael2119** [comment]
please resolve conflict first

**Comment by Alancere** [comment]
Hi @athulvijayanmsft, please update default tag.

File: specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json
Comment (kazrael2119) [inline specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json:5]: I think sdk generation got failed  is because there is different title between here and https://github.com/Azure/azure-rest-api-specs/blob/dce9f88846df8784291f83b9e973e783956b7166/specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2019-10-01/migrate.json#L5

Comment (athulvijayanmsft) [reply specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json:5]: I was iterating there for getting the sdk passed, reverted now.

we are not intenting to change other swagger files in this PR, only the ones under MigrateProjects.

what is the recommendation to set default tag? Does all the individual RP swagger needs to be added in the default tag? But including the remaining swagger files will include them in rest of the checks and that is not intended in this PR.

We are planning to leave the default tag untouched in the PR if possible. I have made changes according to that now. please review this again.

Comment (athulvijayanmsft) [reply specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json:5]: With the current tag settings, we can see sdk-for-go, nettrack passed and there are no breaking changes there. 

But even though `sdk-for-js` looks green in checks, `sdk-for-js` have failed in pipeline due to `authorization` error. 
SDK breaking changes for `sdk-js` are flagged because the pipeline for `sdk-js` failed. Can you let me know why is that?

[`{"level":"error","message":"Create ApiView failed because of authorization failure"}`](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3064270&view=logs&j=47092895-c7c8-567a-c7a6-8c88f3ec178e&t=3de61303-123b-5ab1-40f3-ad043cb08f5f&l=301)

**Comment by Alancere** [comment]
> Hi @athulvijayanmsft, please update default tag.

Hi @athulvijayanmsft, why revert default tag? If so, GO-SDK-CI cannot capture new changes.

File: specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json
Comment (kazrael2119) [reply specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json:5]: @athulvijayanmsft can you update your title to ```Azure Migrate V2```? And revert your readme.md change to this onehttps://github.com/Azure/azure-rest-api-specs/pull/24162/commits/dce9f88846df8784291f83b9e973e783956b7166.

**Comment by kazrael2119** [comment]
your pr has duplicate operations:
[azure-rest-api-specs/specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2019-10-01/migrate.json at f7ee5557ca7cd99ade16373216e81e28e2d3237a · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/blob/f7ee5557ca7cd99ade16373216e81e28e2d3237a/specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2019-10-01/migrate.json#L4862)
and
[azure-rest-api-specs/specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json at f7ee5557ca7cd99ade16373216e81e28e2d3237a · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/blob/f7ee5557ca7cd99ade16373216e81e28e2d3237a/specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json#L2507)

Is the result of calling these two the same?

**Comment by athulvijayanmsft** [comment]
> your pr has duplicate operations: [azure-rest-api-specs/specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2019-10-01/migrate.json at f7ee5557ca7cd99ade16373216e81e28e2d3237a · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/blob/f7ee5557ca7cd99ade16373216e81e28e2d3237a/specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2019-10-01/migrate.json#L4862) and [azure-rest-api-specs/specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json at f7ee5557ca7cd99ade16373216e81e28e2d3237a · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/blob/f7ee5557ca7cd99ade16373216e81e28e2d3237a/specification/migrate/resource-manager/Microsoft.Migrate/MigrateProjects/stable/2023-01-01/hubmigrate.json#L2507)
> 
> Is the result of calling these two the same?

Hi @kazrael2119, The first one is not part of the PR. Please check the files changed tab of the PR.

They are flagged just because it was added in readme default tag.

**Comment by kazrael2119** [comment]
> 

@athulvijayanmsft the duplicate operation will make sdk generation fail, so can you confirm whether the result of calling these two the same? If they are the same, we should be able to avoid this duplicate by adding directive

**Comment by athulvijayanmsft** [comment]
> > 
> 
> @athulvijayanmsft the duplicate operation will make sdk generation fail, so can you confirm whether the result of calling these two the same? If they are the same, we should be able to avoid this duplicate by adding directive

Hi @kazrael2119, the said files under AssessmentProjects (The first link) are taken up in another PR where the local duplicate definitions are replaced by `common-types`. PR for the same [https://github.com/Azure/azure-rest-api-specs/pull/25417](https://github.com/Azure/azure-rest-api-specs/pull/25417)

In the current PR, it is already referencing the `common-types` as per guidelines. 
So, when the other [PR](https://github.com/Azure/azure-rest-api-specs/pull/25417) is merged, there will not be any duplicates.

The scope of this PR is only for MigrateProjects, as AssessmentProjects is a different RP and [PR](https://github.com/Azure/azure-rest-api-specs/pull/25417) for same is under review.

**Comment by athulvijayanmsft** [comment]
Hi @kazrael2119 , can you approve the sdk/PR as there are no real breaking changes here?

**Comment by kazrael2119** [comment]
I think it's best to review your pr after https://github.com/Azure/azure-rest-api-specs/pull/25417 is merged into main and your pr is merged into main branch. After this, sdk-ci-check in this pr may not have duplicate operation and contains your change

**Comment by athulvijayanmsft** [comment]
> @athulvijayanmsft , is this a new service? I found that `hubmigrate.json` is totally different with `migrate.json`. Can you change the default tag to the latest one so that the SDK package can be generated in this PR for review?

Hi @raych1,
This is for a different resourcetype named migrateProjects under Azure.Migrate, while the existing migrate.json is for different resourcetype assessmentProjects. Currently SDK breaking changes-for-js are flagged as the types under existing assessmentProjects are not using `common-types` which is taken care in https://github.com/Azure/azure-rest-api-specs/pull/25417 as a new version and the existing default tag will be deprecated. We request you to approve the changes in this PR and we will take responsibility for any fixes if required during sdk generation in separate PR.

**Comment by raych1** [comment]
@athulvijayanmsft , as they are different resource types and the new swagger file doesn't include the previous resource types, the sdk package name shouldn't be the same one as previous package.

**Comment by athulvijayanmsft** [comment]
> @athulvijayanmsft , as they are different resource types and the new swagger file doesn't include the previous resource types, the sdk package name shouldn't be the same one as previous package.

Hi @raych1,
The package names are different, the old default tag package was `package-migrate-2019-10` which contained assessmentProjects resourcetype.

The current PR package is `tag: package-hubmigrate-2023-01` which has different resourcetype.

Keeping either/both of the package names as the default tag causes checks failure as the resourcetypes and related definitions are different. Also in the current PR we are using `common-types` for few definitions which was local definitions in older resourcetype; which causes duplicate definition flagging.

**Comment by athulvijayanmsft** [comment]
> > @athulvijayanmsft , as they are different resource types and the new swagger file doesn't include the previous resource types, the sdk package name shouldn't be the same one as previous package.
> 
> Hi @raych1, The package names are different, the old default tag package was `package-migrate-2019-10` which contained assessmentProjects resourcetype.
> 
> The current PR package is `tag: package-hubmigrate-2023-01` which has different resourcetype.
> 
> Keeping either/both of the package names as the default tag causes checks failure as the resourcetypes and related definitions are different. Also in the current PR we are using `common-types` for few definitions which was local definitions in older resourcetype; which causes duplicate definition flagging.

Hi @raych1 / @kazrael2119, as the breaking change tests are failing just due to duplicate entries in an older tag which will be deprecated and since the current package follows recommended `common-types`, please approve the sdk breaking change and the PR?

**Comment by kazrael2119** [comment]
approve because of the 2019-10-01 version is going to be deprecated according to service team.

**Comment by athulvijayanmsft** [comment]
> approve because of the 2019-10-01 version is going to be deprecated according to service team.

Thanks @kazrael2119. Can you approve the PR review and add MergeRequested tag?

**Comment by athulvijayanmsft** [comment]
Hi @kazrael2119, can you approve the PR and add merge requested label?

**Comment by kazrael2119** [comment]
> Hi @kazrael2119, can you approve the PR and add merge requested label?

I‘m just responsible for reviewing JS sdk breaking

**Review by raosuhas — APPROVED** [review]

**Pull request merged**
